src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8620,28 +8621,36 @@ private static bool IsAccessibilityModifier(SyntaxKind kind)
             SyntaxListBuilder badBuilder = null;
             for (int i = 0; i < modifiers.Count; i++)
             {
-                switch (modifiers[i].ContextualKind)
+                var modifier = modifiers[i];
+                switch (modifier.ContextualKind)
                 {
                     case SyntaxKind.AsyncKeyword:
                         IsInAsync = true;
                         forceLocalFunc = true;
-                        break;
+                        continue;
                     case SyntaxKind.UnsafeKeyword:
                         forceLocalFunc = true;
-                        break;
-                    case SyntaxKind.StaticKeyword:
+                        continue;
                     case SyntaxKind.ReadOnlyKeyword:
                     case SyntaxKind.VolatileKeyword:
-                        break; // already reported earlier, no need to report again
-                    default:
-                        if (badBuilder == null)
+                        continue; // already reported earlier, no need to report again
+                    case SyntaxKind.StaticKeyword:
+                        if (IsFeatureEnabled(MessageID.IDS_FeatureStaticLocalFunctions))
                         {
-                            badBuilder = _pool.Allocate();
-                            badBuilder.AddRange(modifiers);
+                            continue;
                         }
-                        badBuilder[i] = this.AddError(modifiers[i], ErrorCode.ERR_BadMemberFlag, ((SyntaxToken)modifiers[i]).Text);
+                        modifier = this.AddError(modifier, ErrorCode.ERR_StaticLocalFunctionModifier, new CSharpRequiredLanguageVersion(MessageID.IDS_FeatureStaticLocalFunctions.RequiredVersion()));
```

Comment:
>modifier = this.AddError(modifier, ErrorCode.ERR_StaticLocalFunctionModifier, new CSharpRequiredLanguageVersion(MessageID.IDS_FeatureStaticLocalFunctions.RequiredVersion())); [](start = 24, length = 174)

This doesn't feel right. It looks like this is going to have effect on syntax nodes that goes beyond just an additional diagnostics. For example, an async/unsafe modifiers are likely to be dropped. It is probably worth adjusting the logic so it would never drop original modifiers from the list, only add errors to them at most.

Reply:
The loop should be dropping modifiers, only modifying modifiers that have errors.

---
In reply to: [244893785](https://github.com/dotnet/roslyn/pull/32067#discussion_r244893785) [](ancestors = 244893785)